colour/difference/metamerism.py

```diff
@@ -0,0 +1,296 @@
+"""
+:math:`M_{t}` - Metamerism Index
+====================================================
+
+Define the :math:`M_{t}` metamerism computation objects.
+
+-   :func:`colour.difference.metamerism_index_from_Lab`
+-   :func:`colour.difference.metamerism_index_from_XYZ`
+
+References
+----------
+-   :cite:ISO18314-4_2024 : International Organization for Standardization. (2024).
+    ISO 18314-4:2024. Analytical colorimetry, Part 4: Metamerism index for pairs
+    of samples for change of illuminant (2nd ed., 24 pp.).
+    ISO/TC 256. https://www.iso.org/standard/85116.html
+"""
+
+from __future__ import annotations
+
+import colour
+from colour.hints import (  # noqa: TC001
+    Any,
+    Domain1,
+    Domain100,
+    Literal,
+    LiteralDeltaEMethod,
+    NDArrayFloat,
+)
+from colour.models import XYZ_to_Lab
+from colour.utilities import (
+    as_array,
+    filter_kwargs,
+)
+
+__author__ = "Colour Developers"
+__copyright__ = "Copyright 2013 Colour Developers"
+__license__ = "BSD-3-Clause - https://opensource.org/licenses/BSD-3-Clause"
+__maintainer__ = "Colour Developers"
+__email__ = "colour-developers@colour-science.org"
+__status__ = "Production"
+
+__all__ = [
+    "metamerism_index_from_Lab",
+    "metamerism_index_from_XYZ",
+]
+
+
+def metamerism_index_from_Lab(
```

Comment:
We should probably have `Lab_to_metamerism_index` for naming convention consistency.